core/lib/instrument.c

```diff
@@ -2784,6 +2784,53 @@ dr_get_application_name(void)
 #    endif
 }
 
+dr_error_code_t
+dr_get_error_code(void)
+{
+    dcontext_t *dcontext = get_thread_private_dcontext();
+    return dcontext->client_data->error_code;
+}
+
+int
+dr_num_app_args(void)
+{
+    /* XXX i#2662: Add support for Windows. */
+    return num_app_args();
+}
+
+int
+dr_get_app_args(OUT dr_app_arg_t *args_buf, int buf_size)
+{
+    /* XXX i#2662: Add support for Windows. */
+    return get_app_args(args_buf, (int)buf_size);
+}
+
+const char *
+dr_app_arg_as_cstring(IN dr_app_arg_t *app_arg, char *buf, int buf_size)
+{
+    dcontext_t *dcontext;
+
+    if (app_arg == NULL) {
+        dcontext = get_thread_private_dcontext();
+        dcontext->client_data->error_code = DR_ERROR_INVALID_PARAM;
+        return NULL;
+    }
+
+    switch (app_arg->encoding) {
+    case DR_APP_ARG_CSTR_COMPAT: return (char *)app_arg->start;
+    case DR_APP_ARG_UTF_16:
+        ASSERT_NOT_IMPLEMENTED(false);
```

Comment:
I would have a comment w/ the issue # and say that we have the code to do this (utf16_to_utf8())

Reply:
Done.